extensions/amp-carousel/0.1/amp-carousel.js

```diff
@@ -20,18 +20,18 @@ import {AmpScrollableCarousel} from './scrollable-carousel';
 import {CSS} from '../../../build/amp-carousel-0.1.css';
 import {isExperimentOn} from '../../../src/experiments';
 
-class CarouselSelector {
-
-  constructor(element) {
-    if (element.getAttribute('type') == 'slides') {
-      return new AmpSlideScroll(element);
+class CarouselSelector extends AMP.BaseElement {
+  /** @override */
+  upgradeCallback() {
+    if (this.element.getAttribute('type') == 'slides') {
```

Comment:
The intention is that `upgradeCallback` is called only AFTER attributes are available yet. `upgradeCallback` and only after "connected". If that's not the case - it's a bug and simply needs to be fixed. Under no circumstances, however, we should wait for children.